src/sonic-config-engine/sonic-cfggen

```diff
@@ -47,7 +48,9 @@ def unique_name(l):
 
 def main():
     parser=argparse.ArgumentParser(description="Render configuration file from minigraph data and jinja2 template.")
-    parser.add_argument("-m", "--minigraph", help="minigraph xml file")
+    group = parser.add_mutually_exclusive_group()
+    group.add_argument("-m", "--minigraph", help="minigraph xml file")
+    group.add_argument("-M", "--device-description", help="device description xml file")
```

Comment:
>   group.add_argument("-M", "--device-description", help="device description xml file") [](start = 1, length = 87)

instead of adding a new option, can we treat the device xml as a variant(subset) of minigraph which only contains a subset of information?